### PR TITLE
Add native contract base class: NonfungibleToken

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by Tommo-L** [comment]
The last line will block nodes to relay transactions, even if the node is only one block behind.

https://github.com/neo-project/neo/blob/095d7f7f5c5cad070f35f48e37bb189c5bb2a511/src/neo/Network/P2P/TaskManager.cs#L93-L99

**Comment by vncoelho** [comment]
Good job,@erikzhang.

**Comment by shargon** [comment]
Where it will be used? new token?

File: src/neo/SmartContract/Native/NFTState.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public virtual JObject ToJson()
         {
             return new JObject
             {
+                ["owner"] = Owner.ToString(),
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/NFTState.cs:21]: We don't need owner in json.

**Comment by erikzhang** [comment]
> Where it will be used? new token?

NNS

**Comment by erikzhang** [comment]
Go?

**Review by shargon — APPROVED** [review]
I will add UT when we add the NNS Token

**Review by vncoelho — APPROVED** [review]

**Pull request merged**